src/AgentMulder.Containers.LightInject/Patterns/Register.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.ComponentModel.Composition;
+using AgentMulder.ReSharper.Domain.Patterns;
+using JetBrains.ReSharper.Feature.Services.CSharp.StructuralSearch;
+using JetBrains.ReSharper.Feature.Services.CSharp.StructuralSearch.Placeholders;
+using JetBrains.ReSharper.Feature.Services.StructuralSearch;
+
+namespace AgentMulder.Containers.LightInject.Patterns
+{
+    [Export("ComponentRegistration", typeof(IRegistrationPattern))]
+    public class Register : RegisterWithService
+    {
+        private static readonly IStructuralSearchPattern pattern =
+            new CSharpStructuralSearchPattern("$container$.Register($arguments$)",
+                new ExpressionPlaceholder("container", "global::LightInject.ServiceContainer", true),
+                new ArgumentPlaceholder("arguments", -1, -1));
```

Comment:
Yes, it can, if contain generic argument type. 
e.g.: container.Register<TaskRepository>();
Register pattern is derived from RegisterWithService class, which calls method FromGenericArguments if contain any type argument.

Not sure for max number of argument, from API it takes the most of 4 arguments, what I have found.